# Задача 1

### Подкинули монету N раз. Кол-во случаев, когда выпал орёл, на 10% больше, чем кол-во случаев, когда выпала решка. При каком N мы можем сказать, что монета «нечестная» (орёл и решка выпадают с разной вероятностью)?


Н0(нулевая гипотеза) - монетка честная (выпадает орел или решка с вероятностью 50/50)\
H1(альтернаятивная гипотеза) - монетка нечестная

Может быть, что выпадание орла (например при 10 бросках) будет 6 из 10, тогда может показаться, что монетка нечестная.\
Но это не так, слишком малый набор данных и 6/10 это обычная погрешность при броске. При больших данных можно\ проверить достоверность.

Для того,чтобы отклонить нулевую гипотезу нам нужно p-value менее чем 0.05.



In [87]:
import math
import numpy as np

#функция считающая вероятность выпадания какой нибудь комбинации монеток
def probability(x,n,p): #x-число орлов, n-число бросков, p-предполагаемая вероятность
    return math.factorial(n)/(math.factorial(x)*math.factorial(n-x))*p**x*(1-p)**(n-x)


In [88]:
#Эта функция находит p-value для монетки

def p_value(n,p):
    x = np.round(0.6*n) #Находим вероятность выпадания 60% всех подбрасываний орла
    probability_list = []
    probability_list.append(probability(x=x,n=n,p=p))
    #В этом блоке мы суммируем вероятности более
    #экстремальных событий, например где выпадание орла более 60%
    for i in range(int(n-x)):
        x += 1
        probability_list.append(probability(x=x,n=n,p=0.5))
    #Умножаем на 2, чтобы добавить вероятности чего то такого же редкого
    #Например вероятности выпадания решки 60% и более раз
    return sum(probability_list)*2 


In [89]:
p_val = 1
threshhold = 0.05
n = 10
p = 0.5
while p_val > threshhold:
    p_val = p_value(n=n,p=p)
    n += 1
print(n)

102


Вывод: 102 раза достаточно подбросить монетку,чтобы заподозрить неладное

# Задача 2

### Есть две монеты: «правильная» (с одной стороны орёл, с другой решка) и «неправильная» (орёл на обеих сторонах). Берём одну монету наугад и подкидываем 5 раз. Все 5 раз выпадает орёл. Какова вероятность того, что выбрали «правильную» монету?

Вероятность выпадания орла на неправильной монете 100%, в то время как выпадание орла на правильной 3.125%. Но вопрос звучит не про вероятность выпадания орла или решки, а про то, какую монету мы выбрали, правильную или неправильную. Количество подбрасываний правильной или неправильной монеты не влияет на то, что монета станет другой. Событие происходит в момент выбора монеты. Наугад мы можем выбрать из двух случаев, значит вероятность выбора правильной монеты 50%.

Ответ 50%.